WrightTools/data/_pycmds_large_file.py

```diff
@@ -0,0 +1,196 @@
+    """Create a data object from a single PyCMDS output file.
+
+    Parameters
+    ----------
+    filepath : path-like
+        Path to the .data file
+        Can be either a local or remote file (http/ftp).
+        Can be compressed with gz/bz2, decompression based on file name.
+    name : str or None (optional)
+        The name to be applied to the new data object. If None, name is read
+        from file.
+    parent : WrightTools.Collection (optional)
+        Collection to place new data object within. Default is None.
+    verbose : bool (optional)
+        Toggle talkback. Default is True.
+
+    Returns
+    -------
+    data
+        A Data instance.
+    """
+    filestr = os.fspath(filepath)
+    filepath = pathlib.Path(filepath)
+
+    # header
+    ds = np.DataSource(None)
+    file_ = ds.open(filestr, "rt")
+    headers = tidy_headers.read(file_)
+    # name
+    if name is None:  # name not given in method arguments
+        data_name = headers["data name"]
+    else:
+        data_name = name
+    if data_name == "":  # name not given in PyCMDS
+        data_name = headers["data origin"]
+    # create data object
+    kwargs = {
+        "name": data_name,
+        "kind": "PyCMDS",
+        "source": filestr,
+        "created": headers["file created"],
+    }
+    if parent is not None:
+        data = parent.create_data(**kwargs)
+    else:
+        data = Data(**kwargs)
+    # get axes and scanned variables
+    axes = []
+    for name, identity, units in zip(
+        headers["axis names"], headers["axis identities"], headers["axis units"]
+    ):
+        # points and centers
+        points = np.array(headers[name + " points"])
+        if name + " centers" in headers.keys():
+            centers = headers[name + " centers"]
+        else:
+            centers = None
+        # create
+        axis = {
+            "points": points,
+            "units": units,
+            "name": name,
+            "identity": identity,
+            "centers": centers,
+        }
+        axes.append(axis)
+    shape = tuple([a["points"].size for a in axes])
+    for i, ax in enumerate(axes):
+        sh = [1] * len(shape)
+        sh[i] = len(ax["points"])
+        data.create_variable(
+            name=ax["name"] + "_points", values=np.array(ax["points"]).reshape(sh)
+        )
+        if ax["centers"] is not None:
+            centers = np.array(ax["centers"])
+            sh = list(shape)
+            sh[i] = 1
+            for j, s in enumerate(sh):
+                if centers.size % s:
+                    sh[j] = 1
+            print(f"{ax['name']}_centers")
+            data.create_variable(name=ax["name"] + "_centers", values=centers.reshape(sh))
+    # get assorted remaining things
+    # variables and channels
+    try:
+        signed = iter(headers["channel signed"])
+    except KeyError:
+        signed = itertools.repeat(False)
+    for index, (kind, name) in enumerate(zip(headers["kind"], headers["name"])):
```

Comment:
Only superficial differences to the usual `from_PyCMDS` occur before this point: namely this for loop uses enumerate instead of range and the file is not closed/a full 2D array read. Those lines can be moved to after all headers are parsed, they were placed before so the file could be closed earlier, releiving a small amount of memory, but the effect should be minimal

In this version, this loop only creates variables and channels, does not fill them, filling is done afterwards.
In the original, this loop performs both tasks.
This version also assumes float64 for all fields, the usual version will do whatever genfromtxt gets originally.